src/helpers/analytics/utils.ts

```diff
@@ -1,4 +1,88 @@
+import {
+  HTMLAnchorElement,
+  HTMLButtonElement,
+  HTMLDetailsElement,
+  HTMLElement,
+  HTMLIFrameElement,
+} from '@openstax/types/lib.dom';
 import Cookies from 'js-cookie';
+import { findFirstAncestorOrSelf } from '../../app/domUtils';
+import { isHtmlElement, isNode, isWindow } from '../../app/guards';
+import { UnionToIntersection } from '../../app/types';
 import { disableAnalyticsCookie } from './constants';
 
 export const trackingIsDisabled = () => !!Cookies.get(disableAnalyticsCookie);
+
+type InteractableConfig<T, S> = {
+  events: string[];
+  getState?: (element: T) => S,
+  getStateChange: (element: T, state: S) => string | undefined;
+  getTarget?: (element: T, state: S) => HTMLElement | undefined,
+  match: (element: any) => element is T;
+  state?: S;
+};
+
+// stupid thing to make TS infer the genric type of each config record
+const makeInteractableConfig = <T, S = undefined>(config: InteractableConfig<T, S>) => config;
+
```

Comment:
I tried a few ideas to get rid of the empty function call, but didn't come up with anything better.  Mainly I tried to explicitly type the `interactableElementEvents` array but that got unwieldy.

Reply:
Yea it does suck, but it makes me feel better knowing webpack-terser will remove the empty function call